be/src/exec/parquet_reader.cpp

```diff
@@ -166,6 +176,57 @@ Status ParquetReaderWrap::init_parquet_reader(const std::vector<SlotDescriptor*>
     }
 }
 
+Status ParquetReaderWrap::get_schema(std::vector<SlotDescriptor>* schema) {
+    RETURN_IF_ERROR(_init_parquet_reader());
+
+    auto file_schema = _file_metadata->schema();
+    for (int i = 0; i < file_schema->num_columns(); ++i) {
```

Comment:
`file_schema->num_columns()` returns all leaf columns. 

Maybe we should detect nested types (array/map/struct) as well?

Reply:
That's a good idea. Only the root node of the non-primitive column is treated as VARCHAR now.
This PR only supports the primitive types, and the non-primitive type would be supported in the next PRs.